internal/stolonctl/pkg/kubernetes/status.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/util/node"
+)
+
+// PodStatus represents status of a kubernetes Pod
+type PodStatus struct {
+	// Pod name
+	Name string
+	// IP address of the host to which the pod is assigned
+	HostIP string
+	// IP address allocated to the pod
+	PodIP string
+	// Status of the pod
+	Status string
+	// ReadyContainers - number of ready containers
+	ReadyContainers int
+	// TotalContainers - total number of containers in Pod
+	TotalContainers int
+	// CreationTimestamp is a timestamp representing the server time
+	// when this pod was created
+	CreationTimestamp *metav1.Time
+}
+
+// DeterminePodStatus returns state of the pod, value of total containers
+// and value of ready containers
+func DeterminePodStatus(pod v1.Pod) (string, int, int) {
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+
+	reason := string(pod.Status.Phase)
+	if pod.Status.Reason != "" {
+		reason = pod.Status.Reason
+	}
+
+	initializing := false
+	for i := range pod.Status.InitContainerStatuses {
+		container := pod.Status.InitContainerStatuses[i]
```

Comment:
I think this can be done with an iterator, so instead of splitting this to two lines, it would just be:
```
for i, container := range pod.Status.InitContainerStatuses {
```

More Information: https://gobyexample.com/range